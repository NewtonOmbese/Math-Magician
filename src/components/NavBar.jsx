import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

function NavBar() {
  return (
    <>
      <Navbar>
        <Label>
          Math Magicians
        </Label>
        <Links>
          <Navlinks>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/calculator">Calculator</Link>
            </li>
            <li>
              <Link to="/quote">Quote</Link>
            </li>
          </Navlinks>
        </Links>
      </Navbar>
    </>
  );
}

const Navbar = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 5rem;
`;

const Label = styled.div`
  color: #12ae23;
  font-size: 2rem;
  font-weight: 600;
`;

const Links = styled.div`
  display: flex;
  gap: 2rem;
`;

const Navlinks = styled.ul`
  display: flex;
  gap: 2rem;
  list-style: none;
  li {
    cursor: pointer;
    font-size: 1.2rem;
  }
  a {
    text-decoration: none;
    color: #12ae23;
    font-weight: 600;
    transition: all 0.2s ease;
    &:hover {
      color: #12e223;
      text-decoration: underline;
    }
  }
`;

export default NavBar;
