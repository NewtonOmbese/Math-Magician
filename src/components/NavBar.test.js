import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/jest-globals';
import NavBar from './NavBar';

test('renders the navbar with correct label', () => {
  render(
    <Router>
      <NavBar />
    </Router>,
  );

  const labelElement = screen.getByText('Math Magicians');
  expect(labelElement).toBeInTheDocument();
});

test('renders the navbar with correct links', () => {
  render(
    <Router>
      <NavBar />
    </Router>,
  );

  const homeLink = screen.getByRole('link', { name: 'Home' });
  const calculatorLink = screen.getByRole('link', { name: 'Calculator' });
  const quoteLink = screen.getByRole('link', { name: 'Quote' });

  expect(homeLink).toHaveAttribute('href', '/');
  expect(calculatorLink).toHaveAttribute('href', '/calculator');
  expect(quoteLink).toHaveAttribute('href', '/quote');
});

test('matches snapshot', () => {
  const { container } = render(
    <Router>
      <NavBar />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
