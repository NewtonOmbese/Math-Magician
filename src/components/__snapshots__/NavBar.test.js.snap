// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-bcPKhP dDbnbc"
  >
    <div
      class="sc-grXZZQ iynFHZ"
    >
      Math Magicians
    </div>
    <div
      class="sc-dkjKgF hSQJch"
    >
      <ul
        class="sc-hLclGa cXqXHX"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
